rpcserver.go

```diff
@@ -3158,8 +3159,16 @@ func (r *rpcServer) DescribeGraph(ctx context.Context,
 	err = graph.ForEachChannel(func(edgeInfo *channeldb.ChannelEdgeInfo,
 		c1, c2 *channeldb.ChannelEdgePolicy) error {
 
-		edge := marshalDbEdge(edgeInfo, c1, c2)
-		resp.Edges = append(resp.Edges, edge)
+		// Do not include unannounced channels unless specifically
+		// requested.
+		// Unannounced channels include both private channels as well as
+		// public channels whose authentication proof were not confirmed
+		// yet, hence were not advertized.
+		if includeUnannounced || edgeInfo.AuthProof != nil {
```

Comment:
style nit: To avoid the indentation you can do:
```go
if !includeUnannounced && edgeInfo.AuthProof == nil {
      return nil
}
....append edge...
return nil
```

Reply:
I missed this one, sorry. Why do you think its better to avoid the indentation?

Reply:
I generally like to use "exit early" in case of non-happy path. It's just a style preference though, so if you prefer doing it like this, I don't mind.

In this case it would also have made the diff smaller, which is also a plus ;)

Reply:
On indentation: here it is not that bad, but imagine we had nested conditions we wanted to be satisfied before adding the edge:
```go
if condition1 {
      calculation....
      if condtiondion2 {
            ....more caluclation....
            if condition3 {
                   edges = append(edges, edge)
            }
      }
}
return nil
```

while we could be doing 
```go
if !condition1 {
     return nil
}

calculation....
if !condtiondion2 {
     return nil
}

....more caluclation....
if !condition3 {
    return nil
}
edges = append(edges, edge)
return nil
```

Reply:
Sounds reasonable. The condition is also a bit more clear this way.